src/test/java/hw4/baseclass/steps/assertion/FirstExerciseAssertionStep.java

```diff
@@ -0,0 +1,82 @@
+package hw4.baseclass.steps.assertion;
+
+import hw4.baseclass.utility.GetProperties;
+import hw4.baseclass.pages.FramePage;
+import hw4.baseclass.pages.HomePage;
+import org.openqa.selenium.WebDriver;
+import io.qameta.allure.Step;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.asserts.SoftAssert;
+import java.util.Arrays;
+import java.util.List;
+import static hw4.baseclass.utility.GetProperties.NameOfProperty.EXERCISE_DATA;
+
+public class FirstExerciseAssertionStep {
+
+    private FramePage framePage;
+    private WebDriver driver;
+    private SoftAssert softAssertion;
+    private HomePage homePage;
+    private GetProperties getExerciseDataProperties = new GetProperties(EXERCISE_DATA);
+
+    public FirstExerciseAssertionStep(WebDriver driver, SoftAssert softAssertion,
+                                      WebDriverWait wait) {
+        this.softAssertion = softAssertion;
+        this.driver = driver;
+        framePage = new FramePage(driver);
+        homePage = new HomePage(driver, wait);
+    }
```

Comment:
Here it is acceptable, but in general case could be better to pass pages as params instead of creating them here, because multiple instances of page object (in assert steps and in the test itself) look unnecessary